File: src/main/scala/com/yanns/scala/example3/UserBusiness.scala
package com.yanns.scala.example3

class UserBusiness {

  val userRepository = new UserRepository()

  def isUserAllowed(name: String): Boolean = {

    val allowedUser = for {
      user <- userRepository.findUserByName(name)
      gender <- user.gender
      if gender == "F"
      age <- user.age
      if age >= 18
    } yield user

    allowedUser.isDefined
  }

}

File: src/main/java/com/yanns/java/example3/UserBusiness.java
package com.yanns.java.example3;

public class UserBusiness {

    private final UserRepository userRepository = new UserRepository();

    public boolean isUserAllowed(String name) {
        User user = userRepository.findUserByName(name);
        if (user != null) {
            if (user.getGender() != null && user.getGender().equals("F")) {
                if (user.getAge() != null && user.getAge() >= 18) {
                    return true;
                }
            }
        }
        return false;
    }
}
